opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -994,4 +995,172 @@ mod tests {
+                logs.push((log_record.clone(), instrumentation.clone()));
+            }
+            Ok(())
+        }
+
+        fn shutdown(&mut self) {}
+
+        fn set_resource(&mut self, _resource: &Resource) {}
+    }
+
+    #[test]
+    fn test_simple_processor_sync_exporter_without_runtime() {
+        let exporter = SyncLogExporter {
+            logs: Arc::new(Mutex::new(Vec::new())),
+        };
+        let processor = SimpleLogProcessor::new(Box::new(exporter.clone()));
+
+        let mut record: LogRecord = Default::default();
+        let instrumentation: InstrumentationLibrary = Default::default();
+
+        processor.emit(&mut record, &instrumentation);
+        processor.force_flush().unwrap();
+        processor.shutdown().unwrap();
+
+        assert_eq!(exporter.logs.lock().unwrap().len(), 1);
+    }
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+    async fn test_simple_processor_sync_exporter_with_runtime() {
+        let exporter = SyncLogExporter {
+            logs: Arc::new(Mutex::new(Vec::new())),
+        };
+        let processor = SimpleLogProcessor::new(Box::new(exporter.clone()));
+
+        let mut record: LogRecord = Default::default();
+        let instrumentation: InstrumentationLibrary = Default::default();
+
+        processor.emit(&mut record, &instrumentation);
+        processor.force_flush().unwrap();
+        processor.shutdown().unwrap();
+
+        assert_eq!(exporter.logs.lock().unwrap().len(), 1);
+    }
+
+    #[tokio::test(flavor = "current_thread")]
+    async fn test_simple_processor_sync_exporter_with_current_thread_runtime() {
+        let exporter = SyncLogExporter {
+            logs: Arc::new(Mutex::new(Vec::new())),
+        };
+        let processor = SimpleLogProcessor::new(Box::new(exporter.clone()));
+
+        let mut record: LogRecord = Default::default();
+        let instrumentation: InstrumentationLibrary = Default::default();
+
+        processor.emit(&mut record, &instrumentation);
+        processor.force_flush().unwrap();
+        processor.shutdown().unwrap();
+
+        assert_eq!(exporter.logs.lock().unwrap().len(), 1);
+    }
+
+    #[derive(Debug, Clone)]
+    struct AsyncLogExporter {
+        logs: Arc<Mutex<Vec<(LogRecord, InstrumentationLibrary)>>>,
+    }
+
+    #[async_trait::async_trait]
+    impl LogExporter for AsyncLogExporter {
+        async fn export(&mut self, batch: LogBatch<'_>) -> LogResult<()> {
+            let logs = Arc::clone(&self.logs);
+
+            // Collect batch into an owned Vec to ensure the data can be moved into the tokio::spawn task
+            let batch_owned: Vec<(LogRecord, InstrumentationLibrary)> = batch
+                .iter()
+                .map(|(log_record, instrumentation)| (log_record.clone(), instrumentation.clone()))
+                .collect();
+
+            // Spawn a Tokio task to process the log batch asynchronously
+            tokio::spawn(async move {
```

Comment:
wondering if this is really a scalable way of testing. Should we restrict to testing the features offered by otlp specificially?
otlp + tonic from tokio main, normal main
otlp + request ...
otlp + request blocking

Reply:
Yes but it would be difficult to test that in unit-test with collector. I think we can extend functional test for that. The current test is more generic to cover any exporter which internally uses tokio runtime.

Reply:
agree its best to cover it via the integration test. I did some of them manually and it is not scaleable.

Reply:
Just to be clear, we still keep current generic  test here with runtime - to cover the hang scenario, and panic due to missing runtime.